### The Titanic

![The Titanic](titanic.jpeg)

#### Context

So what do we know already?:

- Most of you will probably have seen the movie by James Cameron (starring Kate Winslet and Leonardo DiCaprio, 1997) which is one of several movies made with the disaster as subject;
- The disaster took place during the night of April 14th, 1912 when the ship hit an iceberg on her mayden voyage out of Southampton (UK) to New York (USA) via Cherbourg (FR) and Queenstown (IRE);
- The loss of lives was 1501 out of a total of 2207 passengers and crew;
- There was a shortage of lifeboats and some of the lifeboats were set afloat whilst not fully occupied.

#### The data

Now we are going to cheat bigtime here, because we will use the dataset(s) prepared for the Kaggle competition. In the wild, out there, you will have to collect the data yourself, often from various sources, messy and incomplete and you will have to spend considerable time searching for data and cleaning the data you found. Estimations vary, but one will probably spend between 60-80% of one's time on searching and cleaning.

Re-use of data, as we are doing here, can be a big timesaver. Hence the importance of "tidy data".

#### Get to know the data

I usually load the data, if it is data in csv format, in my editor and poke aroud a little bit:

In [3]:
!/usr/local/bin/aquamacs /Users/peter/Documents/bootcamps/faector_bc/data/titanic3.csv

All sorts of useful insights can be get this way:

- the name column also contains some extras, like titles: Mr., Mrs., Miss., Master, Rev., Don., etc. as well as the maiden names of female passengers. Passenger 152, for example, is Mrs. Thomas Pearce == Edith Wearne;
- whenever we see two adjacent comma's ',,' we know we are dealing with missing data;
- we might wonder what the column headers "sibsp" and "parch" mean: the number of SIBlings / SPouses abord, the number of PARents / CHildren aboard.

In [4]:
import pandas as pd
import matplotlib.pyplot as plt
% matplotlib inline

In [16]:
df_titanic = pd.read_csv("/Users/peter/Documents/bootcamps/faector_bc/data/titanic3.csv", sep=";")

Now that we have our csv file as a Pandas dataframe, we can use methods on the Pandas "dataframe object" that the library provides:

- [dataframe].shape: Presents the dimensions of the dataframe as a tuple
- [dataframe].head(): Shows the first 5 rows of the dataframe
- [dataframe].describe(): Generates some descriptive statistics
- [dataframe].info(): Generates overview of columns and datatypes and their values

In [17]:
df_titanic.shape

(1310, 14)

In [19]:
df_titanic.head(10)

pclass  survived                                             name     sex  \
0     1.0       1.0                    Allen, Miss. Elisabeth Walton  female   
1     1.0       1.0                   Allison, Master. Hudson Trevor    male   
2     1.0       0.0                     Allison, Miss. Helen Loraine  female   
3     1.0       0.0             Allison, Mr. Hudson Joshua Creighton    male   
4     1.0       0.0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   
5     1.0       1.0                              Anderson, Mr. Harry    male   
6     1.0       1.0                Andrews, Miss. Kornelia Theodosia  female   
7     1.0       0.0                           Andrews, Mr. Thomas Jr    male   
8     1.0       1.0    Appleton, Mrs. Edward Dale (Charlotte Lamson)  female   
9     1.0       0.0                          Artagaveytia, Mr. Ramon    male   

      age  sibsp  parch    ticket      fare    cabin embarked boat   body  \
0      29    0.0    0.0     24160  211,3375       B5        S    2    NaN   
1  0,9167    1.0    2.0    113781  151,5500  C22 C26        S   11    NaN   
2       2    1.0    2.0    113781  151,5500  C22 C26        S  NaN    NaN   
3      30    1.0    2.0    113781  151,5500  C22 C26        S  NaN  135.0   
4      25    1.0    2.0    113781  151,5500  C22 C26        S  NaN    NaN   
5      48    0.0    0.0     19952   26,5500      E12        S    3    NaN   
6      63    1.0    0.0     13502   77,9583       D7        S   10    NaN   
7      39    0.0    0.0    112050    0,0000      A36        S  NaN    NaN   
8      53    2.0    0.0     11769   51,4792     C101        S    D    NaN   
9      71    0.0    0.0  PC 17609   49,5042      NaN        C  NaN   22.0   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON  
5                     New York, NY  
6                       Hudson, NY  
7                      Belfast, NI  
8              Bayside, Queens, NY  
9              Montevideo, Uruguay

In [20]:
df_titanic.describe()

pclass     survived        sibsp        parch        body
count  1309.000000  1309.000000  1309.000000  1309.000000  121.000000
mean      2.294882     0.381971     0.498854     0.385027  160.809917
std       0.837836     0.486055     1.041658     0.865560   97.696922
min       1.000000     0.000000     0.000000     0.000000    1.000000
25%       2.000000     0.000000     0.000000     0.000000   72.000000
50%       3.000000     0.000000     0.000000     0.000000  155.000000
75%       3.000000     1.000000     1.000000     0.000000  256.000000
max       3.000000     1.000000     8.000000     9.000000  328.000000

In [21]:
df_titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1310 entries, 0 to 1309
Data columns (total 14 columns):
pclass       1309 non-null float64
survived     1309 non-null float64
name         1309 non-null object
sex          1309 non-null object
age          1046 non-null object
sibsp        1309 non-null float64
parch        1309 non-null float64
ticket       1309 non-null object
fare         1308 non-null object
cabin        295 non-null object
embarked     1307 non-null object
boat         486 non-null object
body         121 non-null float64
home.dest    745 non-null object
dtypes: float64(5), object(9)
memory usage: 143.4+ KB


What do we want to know?
It might well be you have some questions in advance and want to use the data to see if you can find answers, or it could well be you want to play a little bit with the data in order to come up with questions.

Both approaches suggest the following steps:

- Explore the data (load it, look at it)
- Clean the data (missing values, splitting columns, etc.)
- Plot (try to visualize correlations, insights, ...)
- Assumptions (try to formulate hypotheses, rinse and repeat)

Let's look at the survivors. We know that 38% of the passengers survived the disaster; 68% died. Can we find out somewhat more about the chance of survival?